ElectricCar-ut.cpp

```diff
@@ -0,0 +1,60 @@
+#include "catch2.hpp"
+
+#include "CarHaveMove.hpp"
+#include "ElectricCar.hpp"
+#include "PetrolEngine.hpp"
+#include "WrongEngine.hpp"
+
+SCENARIO("CorrectWorkElectricCar", "Cars"){
+    GIVEN("ElectricCar"){
+        ElectricCar tesla = new ElectricEngine(120, 700);
```

Comment:
implicit conversion. You should rather use:
```suggestion
        auto tesla = ElectricCar{new ElectricEngine(120, 700)};
```